lib/puppet/pops/types/types.rb

```diff
@@ -1263,6 +1467,18 @@ def instance?(o)
         end
       end
 
+      def iterable?
+        true
+      end
+
+      def iterable_type
+        if self == DEFAULT || self == EMPTY
+          PIterableType.new(PTupleType.new([PAnyType::DEFAULT], TUPLE_SIZE))
+        else
+          PIterableType.new(PTupleType.new([@key_type, element_type], TUPLE_SIZE))
```

Comment:
is element_type a typo? or is it element_type() (a call), or should it be @element_type, etc...